src/app/manager-dashboard/reports/reports-myplanet.component.ts

```diff
@@ -149,6 +160,44 @@ export class ReportsMyPlanetComponent implements OnInit {
     this.applyFilters();
   }
 
+  onTimeFilterChange(timeFilter: string) {
+    this.selectedTimeFilter = timeFilter;
+    this.showCustomDateFields = timeFilter === 'custom';
+    if (timeFilter === 'custom') {
+      return;
+    }
+    const now = new Date();
+    let newStartDate: Date;
+    const newEndDate: Date = now;
+
+    switch (timeFilter) {
+      case '24h':
+        newStartDate = new Date(now);
+        newStartDate.setDate(now.getDate() - 1);
+        break;
+      case '7d':
+        newStartDate = new Date(now);
+        newStartDate.setDate(now.getDate() - 7);
+        break;
+      case '30d':
+        newStartDate = new Date(now);
+        newStartDate.setDate(now.getDate() - 30);
+        break;
+      case 'all':
+        newStartDate = this.minDate;
+        break;
+      default:
+        return;
+    }
+    this.startDate = newStartDate;
+    this.endDate = newEndDate;
+    this.reportsForm.patchValue({
+      startDate: newStartDate,
+      endDate: newEndDate,
+    });
+    this.applyFilters();
+  }
+

```

Comment:
Similar to above, since we have a similar block of code in the myPlanet logs, this could be refactored and placed in the reports service and called in both places